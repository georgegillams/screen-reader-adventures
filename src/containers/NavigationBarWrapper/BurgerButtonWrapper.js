import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';

const BurgerButtonWrapper = props => {
  const { children } = props;

  const [isServer, setIsServer] = useState(typeof window === 'undefined');

  useEffect(() => {
    setIsServer(false);
  }, []);

  if (isServer) {
    return (
      <Link href="/sitemap">
        <a {...props} />
      </Link>
    );
  }

  return children;
};

BurgerButtonWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BurgerButtonWrapper;
