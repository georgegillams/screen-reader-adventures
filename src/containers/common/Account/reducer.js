import produce from 'immer';

import { logout, requestVerificationEmail } from './actions';

export const initialState = {
  loggingOut: false,
  logOutError: null,
  logOutResult: null,
  requestingVerificationEmail: false,
  requestVerificationEmailError: null,
  requestVerificationEmailResult: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case logout.REQUEST:
        draft.loggingOut = true;
        draft.logOutError = null;
        break;

      case logout.SUCCESS:
        draft.loggingOut = false;
        draft.logOutResult = payload;
        break;

      case logout.FAILURE:
        draft.loggingOut = false;
        draft.logOutError = payload;
        break;

      case requestVerificationEmail.REQUEST:
        draft.requestingVerificationEmail = true;
        draft.requestVerificationEmailError = null;
        break;

      case requestVerificationEmail.SUCCESS:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailResult = payload;
        break;

      case requestVerificationEmail.FAILURE:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailError = payload;
        break;
    }
  });

export default reducer;
