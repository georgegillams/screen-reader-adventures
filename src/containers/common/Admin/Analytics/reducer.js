import produce from 'immer';

import { load } from './actions';

export const initialState = {
  analytics: null,
  loading: false,
  loadError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.analytics = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;
    }
  });

export default reducer;
