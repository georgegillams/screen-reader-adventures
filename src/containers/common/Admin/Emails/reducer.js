import produce from 'immer';

import { load, resend } from './actions';

export const initialState = {
  emails: null,
  loading: false,
  loadError: null,

  emailToResend: null,
  resending: false,
  resendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.emails = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case resend.TRIGGER:
        draft.emailToResend = payload.emailToResend;
        break;

      case resend.REQUEST:
        draft.resending = true;
        draft.resendError = null;
        break;

      case resend.SUCCESS:
        draft.resending = false;
        break;

      case resend.FAILURE:
        draft.resending = false;
        draft.resendError = payload;
        break;
    }
  });

export default reducer;
