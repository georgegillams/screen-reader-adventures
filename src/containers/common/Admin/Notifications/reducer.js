import produce from 'immer';

import { load, create, remove } from './actions';

export const initialState = {
  notifications: null,
  loading: false,
  loadError: null,

  notificationToCreate: null,
  creating: false,
  createError: null,

  notificationToRemove: null,
  removing: false,
  removeError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.notifications = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case create.TRIGGER:
        draft.notificationToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case remove.TRIGGER:
        draft.notificationToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;
    }
  });

export default reducer;
