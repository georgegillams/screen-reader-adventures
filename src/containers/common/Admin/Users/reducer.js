import produce from 'immer';

import { load, remove, requestMagicLink, create, update } from './actions';

export const initialState = {
  users: null,
  loading: false,
  loadError: null,

  userToRemove: null,
  removing: false,
  removeError: null,

  userToLogIn: null,
  requesting: false,
  requestError: null,

  userToCreate: null,
  creating: false,
  createError: null,

  userToUpdate: null,
  updating: false,
  updateError: null,
  onUpdateSuccessCb: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.users = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case remove.TRIGGER:
        draft.userToRemove = payload;
        break;

      case remove.REQUEST:
        draft.removing = true;
        draft.removeError = null;
        break;

      case remove.SUCCESS:
        draft.removing = false;
        break;

      case remove.FAILURE:
        draft.removing = false;
        draft.removeError = payload;
        break;

      case requestMagicLink.TRIGGER:
        draft.userToLogIn = payload;
        break;

      case requestMagicLink.REQUEST:
        draft.requesting = true;
        draft.requestError = null;
        break;

      case requestMagicLink.SUCCESS:
        draft.requesting = false;
        break;

      case requestMagicLink.FAILURE:
        draft.requesting = false;
        draft.requestError = payload;
        break;

      case create.TRIGGER:
        draft.userToCreate = payload;
        break;

      case create.REQUEST:
        draft.creating = true;
        draft.createError = null;
        break;

      case create.SUCCESS:
        draft.creating = false;
        break;

      case create.FAILURE:
        draft.creating = false;
        draft.createError = payload;
        break;

      case update.TRIGGER:
        draft.userToUpdate = payload.userToUpdate;
        draft.onUpdateSuccessCb = payload.onUpdateSuccessCb;
        break;

      case update.REQUEST:
        draft.updating = true;
        draft.updateError = null;
        break;

      case update.SUCCESS:
        draft.updating = false;
        break;

      case update.FAILURE:
        draft.updating = false;
        draft.updateError = payload;
        break;
    }
  });

export default reducer;
