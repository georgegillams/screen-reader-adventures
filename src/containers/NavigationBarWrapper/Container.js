import React from 'react';
import PropTypes from 'prop-types';
import { NavigationBar } from 'gg-components/NavigationBar';
import NavigationItem from 'components/common/NavigationItem';
import SmallButtonSkeleton from 'gg-components/Skeletons/SmallButtonSkeleton';
import BurgerButtonWrapper from './BurgerButtonWrapper';

import Logo from 'components/Logo';

const NavigationBarWrapper = props => {
  const { authenticatorState } = props;
  const { user } = authenticatorState;

  const accountItem =
    user === undefined ? (
      <SmallButtonSkeleton />
    ) : (
      <NavigationItem name={user ? 'Account' : 'Login'} href={user ? '/account' : '/login'} />
    );

  const adminItem = user && user.admin ? <NavigationItem name="Admin" href="/admin" /> : null;

  const menuItems = [
    <NavigationItem key="home" name="Home" href="/" />,
    adminItem,
    accountItem,
    <NavigationItem key="sitemap" name="Site map" href="/sitemap" />,
  ];

  const date = new Date();
  const isPride = date.getMonth() === 5;

  return (
    <NavigationBar
      menuItems={menuItems}
      logo={<Logo padding={false} pride={isPride} animated />}
      accountMenuItem={accountItem}
      burgerButtonWrapper={BurgerButtonWrapper}
    />
  );
};

NavigationBarWrapper.propTypes = {
  authenticatorState: PropTypes.shape({
    user: PropTypes.object,
  }).isRequired,
};

export default NavigationBarWrapper;
