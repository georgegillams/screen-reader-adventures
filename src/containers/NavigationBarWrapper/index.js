import { memo } from 'react';

import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectState as selectAuthenticatorState } from 'containers/common/Authenticator/selectors';

import NavigationBarWrapper from './Container';

const mapStateToProps = createStructuredSelector({
  authenticatorState: selectAuthenticatorState(),
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect, memo)(NavigationBarWrapper);
