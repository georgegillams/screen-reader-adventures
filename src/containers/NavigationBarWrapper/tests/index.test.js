import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initialState as initialAuthenticatorState } from 'containers/common/Authenticator/reducer';

import configureStore from 'client-utils/common/redux/configure-store';

import NavigationBarWrapperIndex from '../index';
import NavigationBarWrapper from '../Container';

describe('<Authenticator />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBarWrapperIndex />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with default state', () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBarWrapper
          authenticatorState={{
            ...initialAuthenticatorState,
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with user', () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBarWrapper
          authenticatorState={{
            ...initialAuthenticatorState,
            user: { name: 'userName' },
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with null user', () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBarWrapper
          authenticatorState={{
            ...initialAuthenticatorState,
            user: null,
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
