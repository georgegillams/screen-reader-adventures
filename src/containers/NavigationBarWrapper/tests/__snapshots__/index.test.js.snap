// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticator /> should render correctly - index 1`] = `
<div>
  <header
    role="banner"
  >
    <div
      class="navigation-bar__bar-placeholder"
    />
    <div
      class="navigation-bar__container"
    >
      <div
        class="navigation-bar__bar"
      >
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--left"
        >
          <button
            aria-expanded="false"
            aria-label="Menu"
            class="burger-button__outer navigation-bar__burger-button"
            type="button"
          >
            <div
              class="burger-button__inner"
            >
              <div
                class="burger-button__line burger-button__line__one navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__two navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__three navigation-bar__burger-button__line"
              />
            </div>
          </button>
        </div>
        <div
          class="navigation-bar__logo-container"
        >
          <div
            class="logo__container logo__container--no-padding"
          >
            <a
              class="logo__a"
              href="/"
            >
              <h1
                aria-label="Home page"
                class="logo__heading logo__subheading--animated logo__heading--no-padding"
                role="button"
              >
                LOGO
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--rgt"
        >
          <div
            class="skeleton__outer skeleton__button--small"
          >
            <div
              class="skeleton__shimmer"
              style="margin-left: -0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Authenticator /> should render correctly with default state 1`] = `
<div>
  <header
    role="banner"
  >
    <div
      class="navigation-bar__bar-placeholder"
    />
    <div
      class="navigation-bar__container"
    >
      <div
        class="navigation-bar__bar"
      >
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--left"
        >
          <button
            aria-expanded="false"
            aria-label="Menu"
            class="burger-button__outer navigation-bar__burger-button"
            type="button"
          >
            <div
              class="burger-button__inner"
            >
              <div
                class="burger-button__line burger-button__line__one navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__two navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__three navigation-bar__burger-button__line"
              />
            </div>
          </button>
        </div>
        <div
          class="navigation-bar__logo-container"
        >
          <div
            class="logo__container logo__container--no-padding"
          >
            <a
              class="logo__a"
              href="/"
            >
              <h1
                aria-label="Home page"
                class="logo__heading logo__subheading--animated logo__heading--no-padding"
                role="button"
              >
                LOGO
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--rgt"
        >
          <div
            class="skeleton__outer skeleton__button--small"
          >
            <div
              class="skeleton__shimmer"
              style="margin-left: -0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Authenticator /> should render correctly with null user 1`] = `
<div>
  <header
    role="banner"
  >
    <div
      class="navigation-bar__bar-placeholder"
    />
    <div
      class="navigation-bar__container"
    >
      <div
        class="navigation-bar__bar"
      >
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--left"
        >
          <button
            aria-expanded="false"
            aria-label="Menu"
            class="burger-button__outer navigation-bar__burger-button"
            type="button"
          >
            <div
              class="burger-button__inner"
            >
              <div
                class="burger-button__line burger-button__line__one navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__two navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__three navigation-bar__burger-button__line"
              />
            </div>
          </button>
        </div>
        <div
          class="navigation-bar__logo-container"
        >
          <div
            class="logo__container logo__container--no-padding"
          >
            <a
              class="logo__a"
              href="/"
            >
              <h1
                aria-label="Home page"
                class="logo__heading logo__subheading--animated logo__heading--no-padding"
                role="button"
              >
                LOGO
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--rgt"
        >
          <a
            aria-label="Login"
            class="button__a button__outer button__outer--dark-text button__outer--bouncy button__outer--link"
            href="/login"
          >
            <span
              class="button__child"
            >
              Login
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Authenticator /> should render correctly with user 1`] = `
<div>
  <header
    role="banner"
  >
    <div
      class="navigation-bar__bar-placeholder"
    />
    <div
      class="navigation-bar__container"
    >
      <div
        class="navigation-bar__bar"
      >
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--left"
        >
          <button
            aria-expanded="false"
            aria-label="Menu"
            class="burger-button__outer navigation-bar__burger-button"
            type="button"
          >
            <div
              class="burger-button__inner"
            >
              <div
                class="burger-button__line burger-button__line__one navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__two navigation-bar__burger-button__line"
              />
              <div
                class="burger-button__line burger-button__line__three navigation-bar__burger-button__line"
              />
            </div>
          </button>
        </div>
        <div
          class="navigation-bar__logo-container"
        >
          <div
            class="logo__container logo__container--no-padding"
          >
            <a
              class="logo__a"
              href="/"
            >
              <h1
                aria-label="Home page"
                class="logo__heading logo__subheading--animated logo__heading--no-padding"
                role="button"
              >
                LOGO
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navigation-bar__mobile-container navigation-bar__mobile-container--rgt"
        >
          <a
            aria-label="Account"
            class="button__a button__outer button__outer--dark-text button__outer--bouncy button__outer--link"
            href="/account"
          >
            <span
              class="button__child"
            >
              Account
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
